test/manipulation.test.js

```diff
@@ -1622,6 +1622,20 @@ describe('manipulation', function() {
   });
 
   describe('destroy', function() {
+    beforeEach(function resetFixtures(done) {
+      Person.destroyAll(function() {
+        Person.create([
+          {name: 'John'},
+          {name: 'Jane'},
+          {name: 'Mary'},
+        ], done);
```

Comment:
Sharing test data with so many tests is not ideal. When reading the test code and the assertions, it's difficult to understand where the "magic string" names are coming from.

A better solution is to create a helper function for creating an arbitrary list of person instances, and then call that helper from each test.

Also note that new tests can use Promise-based flow control to keep the code simpler.

```js
function givenNamedPeople(namesList) {
  return Promise.all(
    namesList.map(name => Person.create({name})
  );
}

it('should not delete using condition from filter', function() {
  return givenNamedPeople(['John', 'Jane'])
    .then(() => 
      Person.destroyAll({where: {name: 'John'}}).should.be.rejectedWith(/Filter object detected/))
    .then(() => Person.find({fields: 'name'})
    .then(users => { 
      users.should.containDeep([
        // ...
      ]);
    );
});
```

Feel free to keep using callbacks if you prefer, in which case you should use something like [`async.map`](https://github.com/caolan/async/blob/v1.5.2/README.md#map) instead of `Promise.all`.